src/api/app/models/relationship.rb

```diff
@@ -119,6 +121,20 @@ def self.with_groups_and_roles
 
   private
 
+  class << self
+    def add_role(obj, role, opts = {})
+      Relationship::AddRole.new(obj, role, opts).add_role
+    end
+
+    def cache_sequence
```

Comment:
I think these two methods and the constante `RELATIONSHIP_CACHE_SEQUENCE`, make more sense to be on the 8de0cd6b9095e8376c8d75fc6ed98430dcb93522 commit